### PR TITLE
Initial implementation of Night Watch

### PR DESCRIPTION
**Pull request by Lemongrass3110**
* **Addressed Issue(s)**: None

* **Server Mode**: Renewal

* **Description of Pull Request**: 
Taken from #7024

### CODE REVIEW

**Comment by LadyNanuia** [comment]
Theres 2 Sky Emperor skills in this PR that shouldnt be there, we should probobly remove those? 
image

@Lemongrass3110

**Comment by llchrisll** [comment]
Does this PR also include the rebalances?
So I know for the translation project xX

**Comment by Lemongrass3110** [comment]
> Does this PR also include the rebalances? So I know for the translation project xX

@llchrisll no. this is only a copy and cleanup from the original PR by @datawulf in #7024
The rebalances will be in separate PRs to prevent confusion.

**Comment by llchrisll** [comment]
> > Does this PR also include the rebalances? So I know for the translation project xX
> 
> @llchrisll no. this is only a copy and cleanup from the original PR by @datawulf in #7024 The rebalances will be in separate PRs to prevent confusion.

Alright thanks for clarification. Just wanted to make sure ;).
Also different question regarding rebalances will they be done as merged PR or each seperate?

File: src/map/battle.cpp [hunk 1]
```diff
@@ -2591,6 +2591,10 @@ void battle_consume_ammo(map_session_data*sd, int skill, int lv)
 	if (skill) {
 		qty = skill_get_ammo_qty(skill, lv);
 		if (!qty) qty = 1;
+
+		if( skill == NW_MAGAZINE_FOR_ONE && sd != nullptr && sd->weapontype1 == W_GATLING ){
```
Comment (Atemo) [inline src/map/battle.cpp:2595]:
```suggestion
	if (sd == nullptr) {
		return;
	}

	if (skill) {
		qty = skill_get_ammo_qty(skill, lv);
		if (!qty) qty = 1;

		if( skill == NW_MAGAZINE_FOR_ONE && sd->weapontype1 == W_GATLING ){
```

File: src/map/battle.cpp [hunk 2]
```diff
@@ -3427,6 +3461,37 @@ int battle_get_weapon_element(struct Damage* wd, struct block_list *src, struct
...
+
+				if( sc->getSCE( SC_GRENADE_FRAGMENT_4 ) != nullptr ){
+					element = ELE_FIRE;
+				}
+
+				if( sc->getSCE( SC_GRENADE_FRAGMENT_5 ) != nullptr ){
+					element = ELE_DARK;
+				}
+
+				if( sc->getSCE( SC_GRENADE_FRAGMENT_6 ) != nullptr ){
```
Comment (Atemo) [inline src/map/battle.cpp:3490]:
```suggestion
				}
				else if( sc->getSCE( SC_GRENADE_FRAGMENT_2 ) != nullptr ){
					element = ELE_WIND;
				}
				else if( sc->getSCE( SC_GRENADE_FRAGMENT_3 ) != nullptr ){
					element = ELE_EARTH;
				}
				else if( sc->getSCE( SC_GRENADE_FRAGMENT_4 ) != nullptr ){
					element = ELE_FIRE;
				}
				else if( sc->getSCE( SC_GRENADE_FRAGMENT_5 ) != nullptr ){
					element = ELE_DARK;
				}
				else if( sc->getSCE( SC_GRENADE_FRAGMENT_6 ) != nullptr ){
```

**Review by Atemo — COMMENTED** [review]
Quick review

**Pull request merged**